tests/FixerFactoryTest.php

```diff
@@ -387,4 +387,22 @@ private function createFixerMock($name, $priority = 0)
 
         return $fixer;
     }
+
+    /**
+     * @dataProvider provideConflictingFixersRules
+     * @expectedException \UnexpectedValueException
+     * @expectedExceptionMessageRegExp #^The rules contain conflicting fixers:\n#
+     */
+    public function testConflictingFixers(RuleSet $ruleSet)
+    {
+        FixerFactory::create()->registerBuiltInFixers()->useRuleSet($ruleSet);
+    }
+
+    public function provideConflictingFixersRules()
+    {
+        return array(
+            array(new RuleSet(array('short_array_syntax' => true, 'long_array_syntax' => true))),
+            array(new RuleSet(array('long_array_syntax' => true, 'short_array_syntax' => true))),
```

Comment:
Good idea, updated, let me know :)
btw. I think changing the FixerInterface, AbstractFixer and the few conflicting fixers will be easy now (point 2 https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1862#issuecomment-209830341),  let me know if this PR is on track for that.